ArduCopter/AP_Arming.cpp

```diff
@@ -406,6 +411,8 @@ bool AP_Arming_Copter::gps_checks(bool display_failure)
                 if (!mode_requires_gps && fence_requires_gps) {
                     // clarify to user why they need GPS in non-GPS flight mode
                     gcs().send_text(MAV_SEVERITY_CRITICAL,"PreArm: Fence enabled, need 3D Fix");
+                } else if (!mode_requires_gps && check_gps_all_modes) {
+                    gcs().send_text(MAV_SEVERITY_CRITICAL,"PreArm: Checking for GPS on non-GPS modes enabled, need 3D Fix");
```

Comment:
This exceeds the maximum of 50 characters. I'm also not sure we need a different message. If the user has enabled the option to always check GPS then *Need 3D Fix* is standard for him.

Reply:
@OXINARF I found [this](http://ardupilot.org/dev/docs/code-overview-adding-a-new-parameter.html) documentation for adding parameters, but I'm still unsure about how to go about it. Could you point me in the right direction?